tests/api/test_global_cli_context.py

```diff
@@ -84,3 +85,22 @@ def test_forked_context():
         }
 
     assert get_cli_context().connection_context.connection_name is None
+
+
+def test_forked_metrics_spans():
+    outer_metrics = get_cli_context_manager().metrics
+
+    with outer_metrics.start_span("outer_span"):
+        with fork_cli_context() as inner:
+            inner_metrics = inner.metrics
+            with inner_metrics.start_span("inner_span"):
+                pass
+
+    assert outer_metrics != inner_metrics
+    assert outer_metrics.completed_spans != inner_metrics.completed_spans
+
+    assert len(outer_metrics.completed_spans) == len(inner_metrics.completed_spans) == 1
+
+    assert outer_metrics.completed_spans[0][CLIMetricsSpan.NAME_KEY] == "outer_span"
+
+    assert inner_metrics.completed_spans[0][CLIMetricsSpan.NAME_KEY] == "inner_span"
```

Comment:
To me it makes sense that we should reset the spans when we fork the context, since otherwise it could eventually hit the limit if we keep forking the same context and starting spans, but would like to hear if you think otherwise

Reply:
if we start a span before forking, that context will be gone after the fork. it's probably ok, since we most likely don't want to start any spans pre-fork (otherwise it would be a very long span since the interpreter could be very long-lived)